ledger/streaming/client/src/source.rs

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+use displaydoc::Display;
+use futures::{Stream, StreamExt};
+use mc_common::logger::{log, Logger};
+use mc_ledger_streaming_api::{
+    streaming_blocks::{SubscribeRequest, SubscribeResponse},
+    streaming_blocks_grpc::LedgerUpdatesClient,
+    BlockSource, Error, Result,
+};
+use mc_transaction_core::BlockData;
+use std::{convert::TryFrom, time::Duration};
+
+#[derive(Display)]
+pub struct GrpcBlockSource {
+    /// The gRPC client
+    client: LedgerUpdatesClient,
+    /// A logger object
+    logger: Logger,
+}
+
+impl GrpcBlockSource {
+    pub fn new(channel: grpcio::Channel, logger: Logger) -> Self {
+        let client = LedgerUpdatesClient::new(channel);
+        Self { client, logger }
+    }
+}
```

Comment:
Perhaps we should have some kind of sense of uri stored in GRPC block source. At least another field specifying it?  This seems important for enabling logging and record keeping, I am also not %100 certain on this, but I think the ResponderIDs are included in the Merlin Transcripts.

What might be the options for this?

Reply:
Added a URI parameter. I opted for an opaque impl as the parameter, since we haven't converged on where the server will live.